go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -400,6 +400,15 @@ func allocatePodAnnotationWithRollback(
 		hasIPAMClaim = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
 	}
 
+	defer func() {
+		if ipamClaim != nil && claimsReconciler != nil {
+			updatedClaim := claimsReconciler.UpdateIPAMClaimStatus(ipamClaim, podAnnotation, pod.Name, err)
+			if reconcileErr := claimsReconciler.Reconcile(ipamClaim, updatedClaim, ipAllocator); reconcileErr != nil {
+				err = errors.Join(err, fmt.Errorf("failed to reconcile IPAM claim %s/%s: %w", ipamClaim.Namespace, ipamClaim.Name, reconcileErr))
+			}
+		}
+	}()
```

Comment:
Let's just reduce identation, so we don't do if statements nesting.
```suggestion
	defer func() {
		if ipamClaim == nil || claimsReconciler == nil {
		    return 
		}
		updatedClaim := claimsReconciler.UpdateIPAMClaimStatus(ipamClaim, podAnnotation, pod.Name, err)
		if reconcileErr := claimsReconciler.Reconcile(ipamClaim, updatedClaim, ipAllocator); reconcileErr != nil {
			err = errors.Join(err, fmt.Errorf("failed to reconcile IPAM claim %s/%s: %w", ipamClaim.Namespace, ipamClaim.Name, reconcileErr))
		}
	}()
```